nilearn/regions/hierarchical_kmeans_clustering.py

```diff
@@ -247,7 +247,7 @@ def __sklearn_tags__(self):
         if SKLEARN_LT_1_6:
             from nilearn._utils.tags import tags
 
-            return tags()
+            return tags(niimg_like=False)
```

Comment:
Can you explain this change ?

Reply:
yup

This was something that had been forgotten: if you look how tags are defined for more recent sklearn version (line 255), you see that this estimator should not accept niimg like data as input.

But the same had not been done for older version of sklearn and this led to some strange tests failures when testing with them: I eventually tracked down the issue to this.